pkg/lakom/verifysignature/admission.go

```diff
@@ -135,106 +139,262 @@ type handler struct {
+// - extensions.operator.gardener.cloud/Extension: spec.deployment.admission.virtualCluster.helm.ociRepository
+// - extensions.operator.gardener.cloud/Extension: spec.deployment.extension.helm.ociRepository
+func (h *handler) extractExtensionVerificationTargets(ctx context.Context, extension operatorv1alpha1.Extension) ([]verificationTarget, utils.KeyChainReader, error) {
+	var verificationTargets []verificationTarget
+
+	kcr := utils.NewLazyKeyChainReaderFromSecrets(ctx, h.reader, extension.Namespace, []string{}, h.useOnlyImagePullSecrets)
+
+	if extension.Spec.Deployment != nil &&
+		extension.Spec.Deployment.AdmissionDeployment != nil &&
+		extension.Spec.Deployment.AdmissionDeployment.RuntimeCluster != nil &&
+		extension.Spec.Deployment.AdmissionDeployment.RuntimeCluster.Helm != nil &&
+		extension.Spec.Deployment.AdmissionDeployment.RuntimeCluster.Helm.OCIRepository != nil {
+		verificationTargets = append(verificationTargets, verificationTarget{
+			artifactRef: extension.Spec.Deployment.AdmissionDeployment.RuntimeCluster.Helm.OCIRepository.GetURL(),
+			fldPath:     field.NewPath("spec", "deployment", "admissionDeployment", "runtimeCluster", "helm", "ociRepository"),
+		})
+	}
+
+	if extension.Spec.Deployment != nil &&
+		extension.Spec.Deployment.AdmissionDeployment != nil &&
+		extension.Spec.Deployment.AdmissionDeployment.VirtualCluster != nil &&
+		extension.Spec.Deployment.AdmissionDeployment.VirtualCluster.Helm != nil &&
+		extension.Spec.Deployment.AdmissionDeployment.VirtualCluster.Helm.OCIRepository != nil {
+		verificationTargets = append(verificationTargets, verificationTarget{
+			artifactRef: extension.Spec.Deployment.AdmissionDeployment.VirtualCluster.Helm.OCIRepository.GetURL(),
+			fldPath:     field.NewPath("spec", "deployment", "admissionDeployment", "virtualCluster", "helm", "ociRepository"),
+		})
 	}
 
-	for idx, ec := range p.Spec.EphemeralContainers {
-		fldPath := specPath.Child("ephemeralContainers").Index(idx).Child("image")
-		verified, err := h.validateContainerImage(ctx, logger.WithValues("ephemeralContainers", ec.Name), ec.Name, ec.Image, kcr)
+	if extension.Spec.Deployment != nil &&
+		extension.Spec.Deployment.ExtensionDeployment != nil &&
+		extension.Spec.Deployment.ExtensionDeployment.Helm != nil &&
+		extension.Spec.Deployment.ExtensionDeployment.Helm.OCIRepository != nil {
+		verificationTargets = append(verificationTargets, verificationTarget{
+			artifactRef: extension.Spec.Deployment.ExtensionDeployment.Helm.OCIRepository.GetURL(),
+			fldPath:     field.NewPath("spec", "deployment", "extensionDeployment", "helm", "ociRepository"),
+		})
+	}
+
+	return verificationTargets, kcr, nil
+}
+
+// getURL returns the fully-qualified OCIRepository URL of the image.
+func getURL(img *seedmanagementv1alpha1.Image) string {
+	ref := *img.Repository
+
+	if img.Tag != nil {
+		ref = ref + ":" + *img.Tag
```

Comment:
Same as https://github.com/gardener/gardener-extension-shoot-lakom-service/pull/126#discussion_r1949963870, the tag could be a digest.